### PR TITLE
Salesanalyst

### PR DESCRIPTION
**Pull request by beddings81**
Pass first test in iteration 4

### CODE REVIEW

File: spec/sales_analyst_spec.rb [hunk 1]
```diff
@@ -310,4 +310,20 @@
     expect(sales_analyst.invoice_total(1)).to eq(21_067.77)
     expect(sales_analyst.invoice_total(1).class).to eq(BigDecimal)
   end
+
+  it 'can find total revenue by the date' do
+    sales_engine = SalesEngine.from_csv(
+
```
[the remark this answers is not in the document]
Comment (beddings81) [reply spec/sales_analyst_spec.rb:316]: will ignore

[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:316]: If there are a lot of these ignore deleting the line for now

File: spec/sales_engine_spec.rb [hunk 2]
```diff
@@ -86,4 +86,11 @@
 
     expect(se.analyst).to be_a(SalesAnalyst)
   end
+
+  it 'can find all invoices by the date' do
+    se = SalesEngine.from_csv(:invoices => './data/invoices.csv')
+    
+
+    expect(se.find_all_invoices_by_date(Time.parse("2009-02-07"))).to eq([se.invoices.all[0]])
```
[the remark this answers is not in the document]
Comment (beddings81) [reply spec/sales_engine_spec.rb:94]: fixed

File: lib/invoice_repository.rb [hunk 3]
```diff
@@ -31,4 +31,10 @@ def update(id, attributes)
     updated.status = attributes[:status]
     updated.updated_at = Time.now
   end
+
+  def find_all_by_date(date)
+    all.find_all do |invoice|
+    invoice.created_at == date
```
[the remark this answers is not in the document]
Comment (beddings81) [reply lib/invoice_repository.rb:37]: fixed

File: spec/sales_analyst_spec.rb [hunk 4]
```diff
@@ -310,4 +310,20 @@
...
+      :items     => './data/items.csv',
+      :merchants => './data/merchants.csv',
+      :invoices  => './data/invoices.csv',
+      :invoice_items => './data/invoice_items.csv',
+      :transactions => './data/transactions.csv',
+      :customers => './data/customers.csv'
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.total_revenue_by_date(Time.parse("2009-02-07"))).to eq(21_067.77)
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:326]: Pull out the expect argument and have it assigned to an `expected` variable

**Review by Pocketzs — CHANGES_REQUESTED** [review]

**Review by Pocketzs — APPROVED** [review]

**Review by KaraJoHo — APPROVED** [review]

**Review by Kledin85 — APPROVED** [review]
looks good

**Pull request merged**